### PR TITLE
Event/hcai/get attendees list

### PR DESCRIPTION
**Pull request by TCbacon**

### CODE REVIEW

File: APILambda/shared/Cursor.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * Decodes the cursor for the attendees list.
+ *
+ * @param cursor - The encoded cursor string.
+ * @returns Decoded userId and joinDate.
+ */
+export const decodeAttendeesListCursor = (
+  cursor: string
+): { userId: string; joinDate: Date } => {
+  const [encodedUserId, encodedJoinDate] = base64url.decode(cursor).split("|")
+  let joinDate = new Date(defaultJoinDate)
```
Comment (TCbacon) [inline APILambda/shared/Cursor.ts:36]: change encode to decode

File: APILambda/events/getAttendees.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+const mapDatabaseAttendee = (sqlResult: DatabaseAttendeeWithRelation) => {
+  return {
+    id: sqlResult.id,
+    name: sqlResult.name,
+    joinTimestamp: sqlResult.joinTimestamp,
+    profileImageURL: sqlResult.profileImageURL,
+    handle: sqlResult.handle,
+    relations: {
+      youToThem: sqlResult.youToThem,
+      themToYou: sqlResult.themToYou
```
Comment (TCbacon) [inline APILambda/events/getAttendees.ts]: use "not-friends" instead of null for status

File: APILambda/events/getAttendees.test.ts [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+            id: expect.anything(),
+            joinTimestamp: null,
+            name: expect.anything(),
+            profileImageURL: null,
+            relations: { youToThem: null, themToYou: null }
+          })
+        ]),
+        nextPageCursor: expect.stringMatching(lastPageCursorResponse),
+        attendeesCount: expect.anything()
+      })
```
Comment (TCbacon) [inline APILambda/events/getAttendees.test.ts]: check the exact name and id of attendee

File: APILambda/events/getAttendees.test.ts [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+      attendeeToken,
+      eventIds[0],
+      lastPageCursorResp,
+      limit
+    )
+
+    expect(resp).toMatchObject({
+      status: 200,
+      body: expect.objectContaining({
+        attendees: expect.arrayContaining([
```
Comment (TCbacon) [inline APILambda/events/getAttendees.test.ts]: remove objectContaining() and arrayContaining()

File: APILambda/events/getAttendees.test.ts [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+            profileImageURL: null,
+            relations: { youToThem: null, themToYou: null }
+          })
+        ]),
+        nextPageCursor: expect.stringMatching(lastPageCursorResponse),
+        attendeesCount: expect.anything()
+      })
+    })
+  })
+})
```
Comment (TCbacon) [inline APILambda/events/getAttendees.test.ts:287]: Add a test for cursor to be in middle of page

File: APILambda/shared/Cursor.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import base64url from "base64url"
+const defaultJoinDate = "1970-01-01T00:00:00Z"
+
+/**
```
Comment (TCbacon) [inline APILambda/shared/Cursor.ts:3]: Use built in base64 instead of using package

File: APILambda/events/getAttendees.ts [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+        ea.joinTimestamp,
+        MAX(CASE WHEN ur.fromUserId = :userId THEN ur.status END) AS youToThem,
+        MAX(CASE WHEN ur.toUserId = :userId THEN ur.status END) AS themToYou
+        FROM user AS u 
+        INNER JOIN eventAttendance AS ea ON u.id = ea.userId 
+        INNER JOIN event AS e ON ea.eventId = e.id
+        LEFT JOIN userRelations AS ur ON (ur.fromUserId = u.id AND ur.toUserId = :userId)
+                                    OR (ur.fromUserId = :userId AND ur.toUserId = u.id)
+        WHERE e.id = :eventId
+        AND (:nextPageUserIdCursor = 'firstPage' OR (:nextPageUserIdCursor, :nextPageJoinDateCursor) < (u.id, ea.joinTimestamp))
```
Comment (TCbacon) [inline APILambda/events/getAttendees.ts:126]: TODO: use index as cursor instead of userid+joindate

File: APILambda/events/getAttendees.ts [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+  const nextPageJoinDateCursor = hasMoreAttendees
+    ? mappedAttendees[limit - 1].joinTimestamp
+    : null
+
+  const paginatedAttendees = mappedAttendees.slice(0, limit)
+
+  return {
+    nextPageCursor: JSON.stringify({
+      userId: nextPageUserIdCursor,
+      joinDate: nextPageJoinDateCursor
```
Comment (TCbacon) [inline APILambda/events/getAttendees.ts]: return encoded nextPageCursor

**Review by seanim0920 — APPROVED** [review]

**Pull request merged**